src/enable_variable_ops/ngraph_assignadd_op.cc

```diff
@@ -0,0 +1,313 @@
+        BackendManager::GetBackend(ng_backend_name_);
+
+    // Get the Value to be added
+    const Tensor& rhs = context->input(1);
+    // We always return the input ref.
+    context->forward_ref_input_to_ref_output(0, 0);
+
+    string valkey = to_string(ng_graph_id_) + "_" + def().input(1);
+    bool valref_exists = NGraphCatalog::ExistsInOutputCatalog(valkey);
+    shared_ptr<ng::runtime::Tensor> ng_val;
+
+    if (valref_exists) {
+      // Value is from encap
+      Event input_from_catelog("NGAssignAdd: Getting input from catalog",
+                               name().c_str());
+
+      NGRAPH_VLOG(4) << "NGraphAssignAdd::Getting from catalog : " << valkey;
+      ng_val = NGraphCatalog::GetNgTensorFromOutputCatalog(valkey);
+      input_from_catelog.Stop();
+
+      Event::WriteTrace(input_from_catelog);
+    } else {
+      number_of_copies++;
+      copy_log_str << " COPY_INP_VAL[0]";
+      Event input_from_TF("NGAssignAdd: Getting input from TF", name().c_str());
+
+      NGRAPH_VLOG(4) << "NGraphAssignAdd::Getting from TF : " << valkey;
+      TensorShape tfshape = rhs.shape();
+      ng::Shape ng_shape(tfshape.dims());
+      for (int j = 0; j < tfshape.dims(); ++j) {
+        ng_shape[j] = tfshape.dim_size(j);
+      }
+      ng::element::Type ng_element_type;
+      OP_REQUIRES_OK(context, TFDataTypeToNGraphElementType(rhs.dtype(),
+                                                            &ng_element_type));
+      void* tf_src_ptr = (void*)DMAHelper::base(&rhs);
+      ng_val = op_backend->create_tensor(ng_element_type, ng_shape);
+      ng_val->write(tf_src_ptr, 0, ng_val->get_element_count() *
+                                       ng_val->get_element_type().size());
+      Event::WriteTrace(input_from_TF);
+    }
+
+    // Create nGraph Function
+    // Create Input Tensor Vector
+    vector<shared_ptr<ng::runtime::Tensor>> ng_inputs = {ng_tensor_to_assign,
+                                                         ng_val};
+    NGRAPH_VLOG(4) << " Input Tensors Created ";
+
+    std::stringstream signature_ss;
+    for (int i = 0; i < ng_inputs.size(); i++) {
+      auto ngt = ng_inputs[i];
+      for (const auto& x : ng_inputs[i]->get_shape()) {
+        signature_ss << x << ",";
+      }
+      signature_ss << ";";
+    }
+
+    signature_ss << "/";
+    std::string signature = signature_ss.str();
+    NGRAPH_VLOG(4) << " Signature " << signature;
+
+    if (ng_exec_map.find(signature) == ng_exec_map.end()) {
+      Event cache_miss("NGAssignAdd: Cache miss, compiling", name().c_str());
+
+      // create and compile function
+      NGRAPH_VLOG(4) << " Cache miss ";
+      auto V = make_shared<ng::op::Parameter>(
+          ng_tensor_to_assign->get_element_type(),
+          ng_tensor_to_assign->get_shape());
+      auto Val = make_shared<ng::op::Parameter>(ng_val->get_element_type(),
+                                                ng_val->get_shape());
+      auto add = make_shared<ng::op::Add>(V, Val);
```

Comment:
Translate Graph will not see the ng_assign op. It will convert tf_add to ng_add. 

The files are required as the CaptureVariables convert it to ng_assignAdd. So TF needs to know that the kernel and op exists by means of registering the op and kernel. The kernel would never be called.

Reply:
Removed the kernels. Added the registration in new file ngraph_variable_modifiers.cc